task/gpx_ux/locate_me

BarAgam opened the pull request.

gRaytan committed:
start implementing camera_fab
Pay Attention: No butter knofe yet!
Location of the fabs are still incorrect

teamagam1 committed:
Merge branch 'develop' into task/send_image/camera_fab

teamagam1 committed:
layout of FABs placed properly,

teamagam1 committed:
added butterKniife

amazing!

teamagam1 committed:
added taking picture using camera API.

Bitmap result in onActivityResult

teamagam1 committed:
LoctationFetcher updated to support getLastKnownLocation.

onActivityResult from camera saves also time and location.

teamagam1 committed:
added ImageUtil.

now the camera writes the image to temporary cached file with specific uri.

teamagam1 committed:
sending image using PointGeometry instead of LocationSample

teamagam1 committed:
extracting FABs to style

teamagam1 committed:
Merge pull request #19 from teamagam/task/send_image/camera_fab

task/send_image/camera_fab

teamagam1 committed:
Merge remote-tracking branch 'origin/feature/send_image' into task/send_image/camera_fab

teamagam1 committed:
Merge branch 'task/send_image/camera_fab' into task/send_image/take_image

teamagam1 committed:
Toast for uri of the image path

teamagam1 committed:
few CR fixes.

it doesn't create and delete file to get it's URI.

teamagam1 committed:
Merge pull request #23 from teamagam/task/send_image/take_image

task/send_image/take_image

barmako committed:
JS notifies JAVA on cesium ready

barmako committed:
Implemented JS zoomToRectangle function.
Added a listener interface to when MapView is ready for action.
Added a setter for listener interface at GGMapView interface.
Added a getter for GGMapView initialized state.
Implemented that for CesiumMapView.

ViewerFragment reorganized (methods re-ordering by access modifiers)
ViewerFragment implements said interface and:
    Zooms to Israel~ extent when cesium has done loading.
    Attaches SentLocationLayer and UserLocationLayer to GGMapView
    Registers for location updates through receivers

barmako committed:
Merge branch 'feature/send_image' into task/gpx_ux/locate_me

# Conflicts:
#	app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java
#	app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java

barmako committed:
fixed merge issues

barmako committed:
Added "locate me" FAB (in ViewerFragment) that zooms to last known location on click.
Changed FABs icon colors to white.
Removed unused ViewBindings to FABs (Only OnClick binding is needed)
Added config integer to determine viewer's camera height on "locate me" clicks
Added string resource for display on a toast when no last known location is available

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/25" target="_blank"><img src="https://app.review.ninja/52010471/pull/25/badge" alt="ReviewNinja"/>

teamagam1 commented on app/build.gradle:
```diff
@@ -52,9 +52,11 @@ repositories {
             compile 'com.squareup.retrofit2:retrofit:2.0.0'
```
איזה בלאגן.
הוספת פה עוד הרבה קוד ששייך לענפים אחרים. אני מבין שזה נדרש אבל זה מקשה מאוד על הבדיקה של הקוד

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java:
```diff
@@ -107,40 +115,65 @@ public void onReceive(Context context, Intent intent) {
-                    .build();
-            mUsersLocationsLayer.addEntity(point);
+    @OnClick(R.id.locate_me_fab)
+    public void zoomToLastLocation() {
```
consider changing it's name to :zoomToLastKnownLocation

BarAgam replied:
!resolved

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java:
```diff
@@ -107,40 +115,65 @@ public void onReceive(Context context, Intent intent) {
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == Activity.RESULT_OK) {
+            LocationSample imageLocation = LocationFetcher.getInstance(
+                    getActivity()).getLastKnownLocation();
```
I know we all use getActivity().
should we create field mAppContext or something like that and use that instead?

BarAgam replied:
Nope, I think its the way Android wants to play.

barmako committed:
CR fixes:
    Using configurable values to set GGMapView initial extent.
    Resolved race-condition between GGMapView onReady event and it's listener attachment on onCreateView.
    Removed redundant documentation.

barmako committed:
Merge branch 'task/gpx_ux/centralize_map_on_cesium_ip' into task/gpx_ux/locate_me

# Conflicts:
#	app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java
#	app/src/main/res/values/config.xml

barmako committed:
Fixed CR notes and "fine-tuned" last merge
